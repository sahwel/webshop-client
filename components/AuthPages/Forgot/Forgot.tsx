import React from 'react';
import AuthPagesContainer from '../AuthPagesContainer';
import ForgotFooter from './Components/ForgotFooter';
import ForgotHeader from './Components/ForgotHeader';
import ForgotInputs from './Components/ForgotInputs';

const Forgot = () => {
  return (
    <AuthPagesContainer>
      <form className="pt-20 grid gap-16 ">
        <ForgotHeader />
        <ForgotInputs />
        <ForgotFooter />
      </form>
    </AuthPagesContainer>
  );
};

export default Forgot;
