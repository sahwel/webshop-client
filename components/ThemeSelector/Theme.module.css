
@tailwind components;

@layer components {
    .theme {
       @apply rounded-full p-2  flex justify-center items-center
    }
}