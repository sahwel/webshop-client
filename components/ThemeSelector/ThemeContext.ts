/* eslint-disable no-undef */
import { createContext } from 'react';
import { Themes } from './definitions';

export interface ThemeContextModel {
  setTheme(): void;
  getTheme(): Themes;
}

export const ThemeContext = createContext<ThemeContextModel>(null as any);
