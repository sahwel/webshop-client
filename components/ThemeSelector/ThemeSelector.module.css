
@tailwind components;

@layer components {
    .theme_sececltor {
        @apply grid grid-cols-2 z-50 cursor-pointer bg-white absolute right-5  rounded-full border-2 border-theme-border-light
    }  
    
    .theme_sun{
        @apply text-orange-600 
    }

    .theme_moon{
        @apply text-gray-400 rotate-[255deg]
    }

    .theme_moon_highlited{
        @apply bg-[#303841]
    }

    .theme_sun_highlited{
        @apply bg-orange-200
    }
}