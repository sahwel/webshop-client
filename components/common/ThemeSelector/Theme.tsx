import { FunctionComponent, ReactNode } from 'react';
import cl from 'classnames';
import styles from './theme.module.css';

interface ThemeProps {
  children?: ReactNode;
  isHighlighted: boolean;
  highlightedClass: string;
}

const Theme: FunctionComponent<ThemeProps> = ({
  children,
  isHighlighted,
  highlightedClass,
}) => {
  return (
    <div className={cl({ [highlightedClass]: isHighlighted }, styles.theme)}>
      {children}
    </div>
  );
};

export default Theme;
