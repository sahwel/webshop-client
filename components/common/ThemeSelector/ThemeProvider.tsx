import {
  FunctionComponent,
  ReactNode,
  useContext,
  useEffect,
  useLayoutEffect,
  useMemo,
  useState,
} from 'react';
import { ThemeContext, ThemeContextModel } from './ThemeContext';
import { Themes } from './definitions';

interface ThemeProviderProps {
  children?: ReactNode;
}

const ThemeProvider: FunctionComponent<ThemeProviderProps> = ({ children }) => {
  const [currentTheme, setCurrentTheme] = useState<Themes>('light');

  useLayoutEffect(() => {
    // todo: get alternative solution
    const theme = localStorage.theme ? localStorage.theme : 'light';

    if (theme === 'dark') {
      document.documentElement.classList.add('dark');
    } else {
      document.documentElement.classList.remove('dark');
    }
    setCurrentTheme(theme);

    return () => {};
  }, []);
  const contextValue: ThemeContextModel = useMemo(
    () => ({
      setTheme() {
        setCurrentTheme((oldTheme) => {
          const newTheme = oldTheme === 'light' ? 'dark' : 'light';

          if (newTheme === 'dark') {
            document.documentElement.classList.add('dark');
          } else {
            document.documentElement.classList.remove('dark');
          }
          localStorage.theme = newTheme;
          return newTheme;
        });
      },
      getTheme() {
        return currentTheme;
      },
    }),
    [currentTheme]
  );

  return (
    <ThemeContext.Provider value={contextValue}>
      {children}
    </ThemeContext.Provider>
  );
};

export default ThemeProvider;

export function useTheme() {
  const value = useContext(ThemeContext);
  return value;
}
