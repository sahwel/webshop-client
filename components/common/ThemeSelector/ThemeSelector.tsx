import { SunIcon } from '@heroicons/react/outline';
import { MoonIcon } from '@heroicons/react/solid';
import { useCallback, useLayoutEffect, useState } from 'react';
import cl from 'classnames';
import Theme from './Theme';
import styles from './ThemeSelector.module.css';
import { useTheme } from './ThemeProvider';

const ThemeSelector = () => {
  const siteTheme = useTheme();
  const handleChange = useCallback(() => {
    siteTheme.setTheme();
  }, []);

  return (
    <div className={styles.theme_sececltor} onClick={handleChange}>
      <Theme
        highlightedClass={styles.theme_sun_highlited}
        isHighlighted={siteTheme.getTheme() === 'light'}
      >
        <SunIcon className={cl('icon-6', styles.theme_sun)} />
      </Theme>

      <Theme
        highlightedClass={styles.theme_moon_highlited}
        isHighlighted={siteTheme.getTheme() === 'dark'}
      >
        <MoonIcon
          className={cl('icon-6', styles.theme_moon, {
            ['!text-white ']: siteTheme.getTheme() === 'dark',
          })}
        />
      </Theme>
    </div>
  );
};

export default ThemeSelector;
