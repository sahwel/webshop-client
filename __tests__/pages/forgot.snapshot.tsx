import { render, renderHook } from '@testing-library/react';
import {
  mockUseThemeSelector,
  themeSelectorWrapper,
} from '../../testUtils/MockThemeSelector';
import { useTheme } from '@/components/common/ThemeSelector/ThemeProvider';
import React from 'react';
import Forgot from '@/pages/forgot';

React.useContext = mockUseThemeSelector;

describe('forgot snapshot', () => {
  let _result;
  beforeEach(() => {
    const { result } = renderHook(() => useTheme(), {
      wrapper: themeSelectorWrapper,
    });

    _result = result;
  });
  it('renders homepage unchanged', () => {
    const { container } = render(<Forgot />);
    expect(container).toMatchSnapshot();
  });
});
