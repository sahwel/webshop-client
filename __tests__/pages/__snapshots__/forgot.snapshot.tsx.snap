// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`forgot snapshot renders homepage unchanged 1`] = `
<div>
  <div>
    forgot
  </div>
</div>
`;
